test/powershell/Modules/Microsoft.PowerShell.Diagnostics/Export-Counter.Tests.ps1

```diff
@@ -0,0 +1,216 @@
+ # File: Export-Counter.Tests.ps1
+ # Provides Pester tests for the Export-Counter cmdlet.
+ ############################################################################################>
+Describe "Tests for Export-Counter cmdlet" -Tags "CI" {
+
+    BeforeAll {
+        $cmdletName = "Export-Counter"
+
+        . "$PSScriptRoot/CounterTestsCommon.ps1"
+
+        $outputDirectory = $PSScriptRoot
+        $rootFilename = "exportedCounters"
+        $filePath = $null
+        $counterNames = @(
+            (TranslateCounterPath("\Memory\Available Bytes"))
+            (TranslateCounterPath("\Processor(*)\% Processor Time"))
+            (TranslateCounterPath("\Processor(_Total)\% Processor Time"))
+            (TranslateCounterPath("\PhysicalDisk(_Total)\Current Disk Queue Length"))
+            (TranslateCounterPath("\PhysicalDisk(_Total)\Disk Bytes/sec"))
+            (TranslateCounterPath("\PhysicalDisk(_Total)\Disk Read Bytes/sec"))
+        )
+        $counterValues = $null
+
+        # Test the results of Export-Counter by importing the exported
+        # counters and comparing the two sets
+        function CheckExportResults
+        {
+            Test-Path $filePath | Should Be $true
+            $importedCounterValues = Import-Counter $filePath
+
+            CompareCounterSets $counterValues $importedCounterValues
+        }
+
+        # Run a test case
+        function RunTest($testCase)
+        {
+            It "$($testCase.Name)" {
+                $getCounterParams = ""
+                if ($testCase.ContainsKey("GetCounterParams"))
+                {
+                    $getCounterParams = $testCase.GetCounterParams
+                }
+                $counterValues = &([ScriptBlock]::Create("Get-Counter -Counter `$counterNames $getCounterParams"))
+
+                # build up a command
+                $filePath = ""
+                $pathParam = ""
+                $formatParam = ""
+                if ($testCase.ContainsKey("Path"))
+                {
+                    $filePath = $testCase.Path
+                }
+                else
+                {
+                    if ($testCase.ContainsKey("FileFormat"))
+                    {
+                        $formatParam = "-FileFormat $($testCase.FileFormat)"
+                        $filePath = "$outputDirectory/$rootFilename.$($testCase.FileFormat)"
+                    }
+                    else
+                    {
+                        $filePath = "$outputDirectory/$rootFilename.blg"
+                    }
+                }
+                if ($testCase.NoDashPath)
+                {
+                    $pathParam = $filePath
+                }
+                else
+                {
+                    $pathParam = "-Path $filePath"
+                }
+                $cmd = "$cmdletName $pathParam $formatParam -InputObject `$counterValues $($testCase.Parameters) -ErrorAction Stop"
+
```

Comment:
Please add a write-host "Command to run: $cmd". This will help debugging without having to run the test case.